### ISSUE TITLE
Please support AstroNvim

### ISSUE THREAD

**Issue by tsoernes** [topgrade-rs/topgrade#349]
### Which tool is this about? Where is its repository?
AstroNvim https://github.com/AstroNvim/AstroNvim
### Which operating systems are supported by this tool?
Linux
### What should Topgrade do to figure out if the tool needs to be invoked?
I think if there is a AstroNvim gitrepositori in the home directory
```
―― 22:22:35 - Git repositories ―――――――――――――――――――――――――――――――――――――――――――――――――
Pulling /home/torstein/.config/nvim
Failed pulling /home/torstein/.config/nvim
From https://github.com/AstroNvim/AstroNvim
   27bbd0a..039f3a8  v3         -> origin/v3
You are not currently on a branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>
```
### Which exact commands should Topgrade run?
I think `nvim --headless -c ":AstroUpdate"  +qa?`

**Comment by tsoernes** [comment]
I'm not sure if it works. It doesn't give any output:
```
 !  ~/pro/topgrade   349-astro-vim  cargo run                                                                                                                                 (base) 856ms
    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
     Running `target/debug/topgrade`
Due to a design issue with notify-send it could be that topgrade hangs when it's finished.
If this is the case on your system add the --skip-notify flag to the topgrade command or set skip_notify = true in the config file.
If you don't want this message to appear any longer set display_preamble = false in the config file.
For more information about this issue see https://askubuntu.com/questions/110969/notify-send-ignores-timeout and https://github.com/topgrade-rs/topgrade/issues/288.

── 13:26:33 - System update ────────────────────────────────────────────────────
[sudo] password for torstein: 
insync repo                                                                                                                                                    13 kB/s | 2.9 kB     00:00    
Dependencies resolved.
==============================================================================================================================================================================================
 Package                                               Architecture                          Version                                             Repository                              Size
==============================================================================================================================================================================================
Upgrading:
 vim-data                                              noarch                                2:9.0.1262-1.fc36                                   updates                                 23 k
 vim-filesystem                                        noarch                                2:9.0.1262-1.fc36                                   updates                                 18 k
 vim-minimal                                           x86_64                                2:9.0.1262-1.fc36                                   updates                                786 k
 yt-dlp                                                noarch                                2023.01.06-1.fc36                                   updates                                3.5 M
 yt-dlp-bash-completion                                noarch                                2023.01.06-1.fc36                                   updates                                 12 k
 yt-dlp-fish-completion                                noarch                                2023.01.06-1.fc36                                   updates                                 21 k

Transaction Summary
==============================================================================================================================================================================================
Upgrade  6 Packages

Total download size: 4.3 M
Is this ok [y/N]: y
Downloading Packages:
(1/6): vim-data-9.0.1182-1.fc36_9.0.1262-1.fc36.noarch.drpm                                                                                                    46 kB/s |  13 kB     00:00    
(2/6): yt-dlp-bash-completion-2022.11.11-1.fc36_2023.01.06-1.fc36.noarch.drpm                                                                                  23 kB/s | 7.0 kB     00:00    
(3/6): yt-dlp-fish-completion-2022.11.11-1.fc36_2023.01.06-1.fc36.noarch.drpm                                                                                 120 kB/s | 7.4 kB     00:00    
(4/6): vim-filesystem-9.0.1262-1.fc36.noarch.rpm                                                                                                              274 kB/s |  18 kB     00:00    
(5/6): yt-dlp-2022.11.11-1.fc36_2023.01.06-1.fc36.noarch.drpm                                                                                                 1.1 MB/s | 852 kB     00:00    
[DRPM 1/4] vim-data-9.0.1182-1.fc36_9.0.1262-1.fc36.noarch.drpm: done                                                                                                                        
[DRPM 2/4] yt-dlp-bash-completion-2022.11.11-1.fc36_2023.01.06-1.fc36.noarch.drpm: done                                                                                                      
[DRPM 3/4] yt-dlp-fish-completion-2022.11.11-1.fc36_2023.01.06-1.fc36.noarch.drpm: done                                                                                                      
(6/6): vim-minimal-9.0.1262-1.fc36.x86_64.rpm                                                                                                                 1.2 MB/s | 786 kB     00:00    
[DRPM 4/4] yt-dlp-2022.11.11-1.fc36_2023.01.06-1.fc36.noarch.drpm: done                                                                                                                      
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                                         215 kB/s | 1.6 MB     00:07     
Delta RPMs reduced 4.3 MB of updates to 1.6 MB (62.0% saved)
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                      1/1 
  Upgrading        : yt-dlp-2023.01.06-1.fc36.noarch                                                                                                                                     1/12 
  Upgrading        : vim-data-2:9.0.1262-1.fc36.noarch                                                                                                                                   2/12 
  Upgrading        : vim-minimal-2:9.0.1262-1.fc36.x86_64                                                                                                                                3/12 
  Upgrading        : yt-dlp-bash-completion-2023.01.06-1.fc36.noarch                                                                                                                     4/12 
  Upgrading        : yt-dlp-fish-completion-2023.01.06-1.fc36.noarch                                                                                                                     5/12 
  Upgrading        : vim-filesystem-2:9.0.1262-1.fc36.noarch                                                                                                                             6/12 
  Cleanup          : yt-dlp-bash-completion-2022.11.11-1.fc36.noarch                                                                                                                     7/12 
  Cleanup          : yt-dlp-fish-completion-2022.11.11-1.fc36.noarch                                                                                                                     8/12 
  Cleanup          : yt-dlp-2022.11.11-1.fc36.noarch                                                                                                                                     9/12 
  Cleanup          : vim-filesystem-2:9.0.1182-1.fc36.noarch                                                                                                                            10/12 
  Cleanup          : vim-minimal-2:9.0.1182-1.fc36.x86_64                                                                                                                               11/12 
  Cleanup          : vim-data-2:9.0.1182-1.fc36.noarch                                                                                                                                  12/12 
  Running scriptlet: vim-data-2:9.0.1182-1.fc36.noarch                                                                                                                                  12/12 
  Verifying        : vim-data-2:9.0.1262-1.fc36.noarch                                                                                                                                   1/12 
  Verifying        : vim-data-2:9.0.1182-1.fc36.noarch                                                                                                                                   2/12 
  Verifying        : vim-filesystem-2:9.0.1262-1.fc36.noarch                                                                                                                             3/12 
  Verifying        : vim-filesystem-2:9.0.1182-1.fc36.noarch                                                                                                                             4/12 
  Verifying        : vim-minimal-2:9.0.1262-1.fc36.x86_64                                                                                                                                5/12 
  Verifying        : vim-minimal-2:9.0.1182-1.fc36.x86_64                                                                                                                                6/12 
  Verifying        : yt-dlp-2023.01.06-1.fc36.noarch                                                                                                                                     7/12 
  Verifying        : yt-dlp-2022.11.11-1.fc36.noarch                                                                                                                                     8/12 
  Verifying        : yt-dlp-bash-completion-2023.01.06-1.fc36.noarch                                                                                                                     9/12 
  Verifying        : yt-dlp-bash-completion-2022.11.11-1.fc36.noarch                                                                                                                    10/12 
  Verifying        : yt-dlp-fish-completion-2023.01.06-1.fc36.noarch                                                                                                                    11/12 
  Verifying        : yt-dlp-fish-completion-2022.11.11-1.fc36.noarch                                                                                                                    12/12 

Upgraded:
  vim-data-2:9.0.1262-1.fc36.noarch                     vim-filesystem-2:9.0.1262-1.fc36.noarch               vim-minimal-2:9.0.1262-1.fc36.x86_64       yt-dlp-2023.01.06-1.fc36.noarch      
  yt-dlp-bash-completion-2023.01.06-1.fc36.noarch       yt-dlp-fish-completion-2023.01.06-1.fc36.noarch      

Complete!

── 13:27:03 - yadm ─────────────────────────────────────────────────────────────
Enter passphrase for key '/home/torstein/.ssh/github': 
Already up to date.

── 13:27:09 - Git repositories ─────────────────────────────────────────────────
Pulling /home/torstein/.config/nvim
Failed pulling /home/torstein/.config/nvim
You are not currently on a branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

Git repositories failed: 
   0: You are not currently on a branch.
      Please specify which branch you want to merge with.
      See git-pull(1) for details.

          git pull <remote> <branch>

   0: 

Location:
   src/steps/git.rs:40

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
Retry? (y)es/(N)o/(s)hell/(q)uit

── 13:27:17 - oh-my-fish ───────────────────────────────────────────────────────
Oh My Fish is up to date.
You are now using Oh My Fish version 7.
Updating https://github.com/oh-my-fish/packages-main master... Done!
fish is already up-to-date.
bobthefish is already up-to-date.
default is already up-to-date.

── 13:27:20 - TLDR ─────────────────────────────────────────────────────────────
Updated cache for 0 entries

── 13:27:24 - rustup ───────────────────────────────────────────────────────────
info: checking for self-updates
  rustup unchanged - 1.25.2

info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
info: checking for self-updates

  stable-x86_64-unknown-linux-gnu unchanged - rustc 1.67.0 (fc594f156 2023-01-24)

info: cleaning up downloads & tmp directories

── 13:27:24 - Doom Emacs ───────────────────────────────────────────────────────
! Failed to locate a ':completion +childframe' module
! Failed to locate a ':completion +icons' module
! Failed to locate a ':ui +lsp' module
> Preparing to upgrade Doom Emacs and its packages...
x There was an unexpected runtime error
  Message: Failed to fetch from upstream
  Backtrace:
    (error "Failed to fetch from upstream")
    (or (= 0 (car (setq result (sh! "git" "fetch" "--force" "--tags" doom-upgrade-remote (format "%s:%s" branch target-remote))))) (error "Failed to fetch from upstream"))
    (let (result) (or (= 0 (car (sh! "git" "remote" "add" doom-upgrade-remote doom-upgrade-url))) (error "Failed to add %s to remotes" doom-upgrade-remote)) (or (= 0 (car (setq result (s...
    (unwind-protect (let (result) (or (= 0 (car (sh! "git" "remote" "add" doom-upgrade-remote doom-upgrade-url))) (error "Failed to add %s to remotes" doom-upgrade-remote)) (or (= 0 (car...
    (let* ((branch (replace-regexp-in-string "^\\(?:[^/]+/[^/]+/\\)?\\(.+\\)\\(?:~[0-9]+\\)?$" "\\1" (cdr (sh! "git" "name-rev" "--name-only" "HEAD")))) (target-remote (format "%s_%s" do...
    (let ((default-directory doom-emacs-dir) process-file-side-effects) (doom-print (doom-print--format (doom-print--class 'start "Preparing to upgrade Doom Emacs and its packages...")))...
    (doom-cli-upgrade #s(doom-cli-context (25564 64940 700216 139000) 124887 1 (190 . 48) nil nil "doom" nil nil ("doom" "upgrade") nil ("upgrade") (("--jobs") ("-j") ("--packages") ("-p...
    (cond (packages? (let ((recipe (doom-cli-context-get context 'straight-recipe))) (if (and recipe (not (equal recipe (doom-upgrade--get-straight-recipe)))) (progn (doom-print (doom-pr...
    (let* ((force? (progn (or (progn (and (memq (type-of context) cl-struct-doom-cli-context-tags) t)) (signal 'wrong-type-argument (list 'doom-cli-context context))) (aref context 6))) ...
    (let ((packages? (cdr (assq 'packages? alist))) (jobs (cdr (assq 'jobs alist))) (context (cdr (assq 'context alist)))) (let* ((force? (progn (or (progn (and (memq (type-of context) c...
    ((closure (t) (cli alist) (let ((packages? (cdr (assq 'packages? alist))) (jobs (cdr (assq 'jobs alist))) (context (cdr (assq 'context alist)))) (let* ((force? (progn (or (progn (and...
    (funcall (closure (t) (cli alist) (let ((packages? (cdr (assq 'packages? alist))) (jobs (cdr (assq 'jobs alist))) (context (cdr (assq 'context alist)))) (let* ((force? (progn (or (pr...
  ! Wrote extended backtrace to ~/.emacs.d/.local/logs/cli.doom.230203132724.124887.error
Emacs failed: 
   0: Command failed: `/home/torstein/.emacs.d/bin/doom upgrade`
   1: `/home/torstein/.emacs.d/bin/doom` failed: exit status: 255

Location:
   src/steps/emacs.rs:77

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
Retry? (y)es/(N)o/(s)hell/(q)uit

── 13:27:33 - Conda ────────────────────────────────────────────────────────────
Collecting package metadata (current_repodata.json): done
Solving environment: done

# All requested packages already installed.


── 13:27:53 - pip3 ─────────────────────────────────────────────────────────────
Requirement already satisfied: pip in /home/torstein/.local/lib/python3.9/site-packages (23.0)

── 13:27:56 - stack ────────────────────────────────────────────────────────────
Current Stack version: 2.9.3, available download version: 2.9.3
Skipping binary upgrade, you are already running the most recent version

── 13:27:56 - Neovim ───────────────────────────────────────────────────────────
Plugins upgraded

── 13:28:04 - Node Package Manager ─────────────────────────────────────────────

── 13:28:04 - Flatpak User Packages ────────────────────────────────────────────
Looking for updates…
Nothing to do.

── 13:28:05 - Flatpak System Packages ──────────────────────────────────────────
Looking for updates…
Nothing to do.

── 13:28:06 - snap ─────────────────────────────────────────────────────────────
All snaps up to date.

── 13:28:07 - Firmware upgrades ────────────────────────────────────────────────
Firmware metadata last refresh: 21 hours ago. Use --force to refresh again.
Devices with no available firmware updates: 
 • UEFI Device Firmware
 • UEFI Device Firmware
Devices with the latest available firmware version:
 • Embedded Controller
 • Intel Management Engine
 • MZVLB1T0HBLR-000L7
 • Prometheus
 • Prometheus IOTA Config
 • System Firmware
 • UEFI dbx
No updates available

── 13:28:08 - Summary ──────────────────────────────────────────────────────────
System update: OK
config-update: OK
yadm: OK
Git repositories: FAILED
oh-my-fish: OK
TLDR: OK
rustup: OK
Emacs: FAILED
conda: OK
pip3: OK
stack: OK
Neovim: OK
Flatpak: OK
snap: OK
Firmware upgrades: OK
 !  ~/pro/topgrade   349-astro-vim  git status                                                                                                                                 (base) 1.6m
On branch 349-astro-vim
Your branch is up to date with 'dottodev/349-astro-vim'.

nothing to commit, working tree clean
```

**Comment by DottoDev** [comment]
Could you set `RUST_BACKTRACE=full ` and then send in the output again?

**Comment by DottoDev** [comment]
Could you check when astrovim has an update and then run topgrade. Then you should be able to see if it works. But will merge for now.
